rusqlite version is not fixed - results in rust crate being unusable

I can not install this crate with:
```toml
deltachat = { git = "https://github.com/deltachat/deltachat-core-rust", tag="1.75.0"}
```
Because of the not fixed version of `rusqlite` in cargo.toml:
```toml
[patch.crates-io]
rusqlite = { git = "https://github.com/rusqlite/rusqlite", branch="master" }
```
I get a newer version of `rusqlite` that is incompatible (does not have some enums we require anymore):

```
$ cargo check
    Blocking waiting for file lock on build directory
    Checking deltachat v1.74.0 (https://github.com/deltachat/deltachat-core-rust?tag=1.74.0#99ddce6c)
error[E0599]: no variant or associated item named `SQLITE_DBCONFIG_RESET_DATABASE` found for enum `DbConfig` in the current scope
   --> ~/.cargo/git/checkouts/deltachat-core-rust-632648ad67f90089/99ddce6/src/sql.rs:139:38
    |
139 |         conn.set_db_config(DbConfig::SQLITE_DBCONFIG_RESET_DATABASE, true)
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `DbConfig`

error[E0599]: no variant or associated item named `SQLITE_DBCONFIG_RESET_DATABASE` found for enum `DbConfig` in the current scope
   --> ~/.cargo/git/checkouts/deltachat-core-rust-632648ad67f90089/99ddce6/src/sql.rs:143:38
    |
143 |         conn.set_db_config(DbConfig::SQLITE_DBCONFIG_RESET_DATABASE, false)
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `DbConfig`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0599`.
error: could not compile `deltachat`
```
(yes this log is from 1.74, I also tried it, because 1.75 did not work)

I'll try to manually edit my projects's lock file, but would be only a bandaid - the problem still needs to be fixed in core's cargo toml.

**Update**: Workaround:
cargo.toml:
```toml
# this is required until https://github.com/deltachat/deltachat-core-rust/issues/3080 is solved
[patch.crates-io]
rusqlite = {git="https://github.com/rusqlite/rusqlite", rev = "ddb7141c6dee4b8956af85b2e4a01a28e5fdbacc" }
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,9 +12,6 @@ debug = 0
 [profile.release]
 lto = true
 
-[patch.crates-io]
-rusqlite = { git = "https://github.com/rusqlite/rusqlite", branch="master" }
-
 [dependencies]
 deltachat_derive = { path = "./deltachat_derive" }
 
@@ -53,10 +50,10 @@ pgp = { version = "0.7", default-features = false }
 pretty_env_logger = { version = "0.4", optional = true }
 quick-xml = "0.22"
 r2d2 = "0.8"
-r2d2_sqlite = "0.19"
+r2d2_sqlite = "0.20"
 rand = "0.7"
 regex = "1.5"
-rusqlite = { version = "0.26", features = ["sqlcipher"] }
+rusqlite = { version = "0.27", features = ["sqlcipher"] }
 rust-hsluv = "0.1"
 rustyline = { version = "9", optional = true }
 sanitize-filename = "0.3"

